### PR TITLE
Introduce HISTORY role

### LINKED ISSUE #7444: history, historyViewRevision and historyCompareRevisions should have his own permission map `HISTORY`
## Feature Request

When doing https://github.com/sonata-project/SonataAdminBundle/pull/7443, I saw that `history`, `historyViewRevision` and `historyCompareRevisions` was using the `EDIT` permission. 

I was kinda surprise the role `ROLE_MYADMIN_HISTORY` didn't exist ; IMHO it should.

It's because you have the right to edit some field of some entity that you have the right to see all the modification made by others people. And you can also have the right to see the history without the right to edit.

WDYT @sonata-project/contributors ?

### PR DESCRIPTION
**Pull request by VincentLanglet**
## Subject

I am targeting this branch, because there is no real BC way to do it.
It will be a BC-break only for revisions users (with EntityAuditBundle).

Closes #7444.

## Changelog

```markdown
### Added
- New role `ROLE_MYADMIN_HISTORY` to access to the history actions

### Changed
- `AbstractAdmin::getAccessMapping` visibility

### Removed
- Access to the history actions with the role `ROLE_MYADMIN_EDIT` 
- `AdminInterface::getAccessMapping` method
```

### CODE REVIEW

**Comment by VincentLanglet** [comment]
The second part of the issue (https://github.com/sonata-project/SonataAdminBundle/issues/7444#issuecomment-909481598) can wait for 5.X

File: src/Admin/AbstractAdmin.php [hunk 1]
```diff
@@ -1731,11 +1724,6 @@ final public function getListMode(): string
         return $this->getRequest()->getSession()->get(sprintf('%s.list_mode', $this->getCode()), 'list');
     }
 
-    final public function getAccessMapping(): array
```
Comment (core23) [inline src/Admin/AbstractAdmin.php:1734]: You should mention this inside the CHANGELOG.

**Review by core23 — DISMISSED** [review]

File: src/Admin/AbstractAdmin.php (hunk 1 above)
Comment (VincentLanglet) [reply src/Admin/AbstractAdmin.php:1734]: Done

**Review by jordisala1991 — APPROVED** [review]

**Review by dmaicher — APPROVED** [review]

**Comment by jordisala1991** [comment]
Thanks @VincentLanglet

**Pull request merged**